Detectors/Base/include/DetectorsBase/Vector.h

```diff
@@ -0,0 +1,395 @@
+#ifndef O2_BASE_VECTOR_H
+#define O2_BASE_VECTOR_H
+
+#include <TTree.h>
+#include <TBranch.h>
+#include <TBranchElement.h>
+#include <TBuffer.h>
+#include <TObject.h>
+#include <memory>
+#include <algorithm>
+#include <type_traits>
+#include <stdexcept>
+#include <iostream>
+#include <FairLogger.h>
+
+
+namespace o2 {
+namespace Base {
+
+ template <class T, class H>
+   class Vector: public TObject {
```

Comment:
I think the name "Vector" for this class is a bit too unspecific and easily confusable with std::vector etc. Can we consider having a more specialized name?